### PR TITLE
GUVNOR-2971 : [GSS](6.4.x)Artifact repository upload functionality allows malicious code injection

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/GUVNOR-2971

### CODE REVIEW

**Comment by Rikkola** [comment]
@karreiro or who ever feels like they want to review, please take a look.

File: guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/helpers/HttpPostHelper.java [hunk 1]
```diff
@@ -208,9 +207,13 @@ private String uploadPom( final FormData formData ) throws IOException {
...
+                          pbe);
+                return UPLOAD_UNABLE_TO_PARSE_POM;
             } catch (MavenEmbedderException mee) {
-                return mee.getMessage();
+                log.error(mee.getMessage(),
+                          mee);
+                return UPLOAD_UNABLE_TO_PARSE_POM;
             }catch (Exception e) {
                 return UPLOAD_UNABLE_TO_PARSE_POM;
             }
```
Comment (psiroky) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/helpers/HttpPostHelper.java]: Is there a specific reason to _not_ log the general `Exception`? If not, the whole catch block could be simplified to just
```
catch (Exception ex) {
    log.error("Can no parse the provided POM", ex); // or some better log message
    return UPLOAD_UNABLE_TO_PARSE_POM;
}
```
which also brings different question: should we even catch these general `Exception`s? Shouldn't they be propaged and caught the the top-level `UncaughtExceptionHandler` (or how we call it)?

Comment (Rikkola) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/helpers/HttpPostHelper.java]: This is used by the file upload servlet so the UncaughtExceptionHandler might not pick it up, but true we should probably just capture any exception.

**Comment by Rikkola** [comment]
@karreiro Updated with the exception change @psiroky mentioned. Functionality is still the same.

**Comment by Rikkola** [comment]
jenkins please retest this

**Pull request merged**